mesh_aws_client/mesh_check_send_parameters_application.py

```diff
@@ -0,0 +1,143 @@
+"""
+Module for MESH API functionality for step functions
+"""
+from http import HTTPStatus
+import os
+from math import ceil
+import boto3
+from aws_lambda_powertools.utilities.data_classes import EventBridgeEvent
+from spine_aws_common import LambdaApplication
+from spine_aws_common.utilities import human_readable_bytes
+from mesh_aws_client.mesh_common import MeshCommon, SingletonCheckFailure
+
+
+def calculate_chunks(file_size, chunk_size):
+    """Helper for number of chunks"""
+    chunks = ceil(file_size / chunk_size)
+    do_chunking = chunks > 1
+    return (do_chunking, chunks)
+
+
+class MeshCheckSendParametersApplication(LambdaApplication):
+    """MESH API Lambda for sending a message"""
+
+    def __init__(self, additional_log_config=None, load_ssm_params=False):
+        """Initialise variables"""
+        super().__init__(additional_log_config, load_ssm_params)
+        self.environment = self.system_config.get("Environment", "default")
+        self.chunk_size = None
+        self.my_step_function_name = None
+
+    def _get_internal_id(self):
+        """Override to stop crashing when getting from non-dict event"""
+        return self._create_new_internal_id()
+
+    def process_event(self, event):
+        return EventBridgeEvent(event)
+
+    def initialise(self):
+        self.chunk_size = int(
+            self.system_config.get("CHUNK_SIZE", MeshCommon.DEFAULT_CHUNK_SIZE)
+        )
+        # TODO figure out better way to do this:
+        self.my_step_function_name = f"{self.environment}-mesh-send-message"
```

Comment:
Maybe use an environment variable here, and that can be set by Terraform easily enough to the value of the step function name?

Reply:
I'm more thinking in line of getting the step function name from the execution context, I think I have an idea how to do this